testing/ceph/APKBUILD

```diff
@@ -114,59 +117,69 @@ _sysconfdir=/etc
 _udevrulesdir=/etc/udev/rules.d
 _python_sitelib=/usr/lib/python2.7/site-packages
 
-
-_builddir="$srcdir"/$pkgname-$pkgver
-
 build() {
-   cd "$_builddir"
-   mkdir build
-   cd build
-   cmake .. \
-	   -DCMAKE_INSTALL_PREFIX=$_prefix \
-	   -DCMAKE_INSTALL_LIBDIR=$_libdir \
-	   -DCMAKE_INSTALL_LIBEXECDIR=$_libexecdir \
-	   -DCMAKE_INSTALL_LOCALSTATEDIR=$_localstatedir \
-	   -DCMAKE_INSTALL_SYSCONFDIR=$_sysconfdir \
-	   -DCMAKE_INSTALL_DOCDIR=$_docdir/ceph \
-	   -DCMAKE_INSTALL_MANDIR=$_mandir \
-	   -DWITH_REENTRANT_STRSIGNAL=ON \
-	   -DWITH_THREAD_SAFE_RES_QUERY=ON \
-	   -DWITH_MANPAGE=OFF \
-	   -DWITH_PYTHON3=OFF \
-	   -DWITH_TESTS=$WITH_TESTS \
-	   || return 1
-   make
+	export CEPH_BUILD_VIRTUALENV=$builddir
+
+	mkdir -p $builddir/build
+	cd $builddir/build
+
+	if [ -n "$MAKE_CHECK" ] || [ -n "$CEPH_TEST_PKG" ]; then
+		local _with_tests=ON
+	fi
+
+	cmake .. \
+		-DCMAKE_INSTALL_PREFIX=$_prefix \
+		-DCMAKE_INSTALL_LIBDIR=$_libdir \
+		-DCMAKE_INSTALL_LIBEXECDIR=$_libexecdir \
+		-DCMAKE_INSTALL_LOCALSTATEDIR=$_localstatedir \
+		-DCMAKE_INSTALL_SYSCONFDIR=$_sysconfdir \
+		-DCMAKE_INSTALL_DOCDIR=$_docdir/ceph \
+		-DCMAKE_INSTALL_MANDIR=$_mandir \
+		-DWITH_REENTRANT_STRSIGNAL=ON \
+		-DWITH_THREAD_SAFE_RES_QUERY=ON \
+		-DWITH_MANPAGE=ON \
+		-DWITH_PYTHON3=OFF \
+		-DWITH_LTTNG=OFF \
+		-DWITH_SYSTEM_BOOST=ON \
+		-DWITH_EMBEDDED=OFF \
+		-DWITH_TESTS=${_with_tests:-OFF} \
+		|| return 1
+	make -j${JOBS:-2} || return 1
+
+	if [ -n "$MAKE_CHECK" ]; then
+		ctest -j${JOBS:-2}
+	fi
 }
 
 package() {
-	cd "$_builddir"
-	cd build
-	make DESTDIR="$pkgdir" install
+	cd $builddir/build
+	make DESTDIR=$pkgdir install || return 1
 	cd ..
 
-	rm -f "$pkgdir"$_sysconfdir/init.d/ceph
+	rm -f $pkgdir$_sysconfdir/init.d/ceph
```

Comment:
Why have you removed quotes around pkgdir? Variables that contain externally provided paths (like `builddir`, `pkgdir`, `srcdir`, …) should be quoted.